src/typing/flow_js.ml

```diff
@@ -5555,10 +5555,10 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       add_output cx ~trace FlowError.(EInternal (loc, ShadowWriteComputed))
 
     (* LookupT is a non-strict lookup *)
-    | (ObjProtoT _ |
-       FunProtoT _ |
-       DefT (_, MixedT (Mixed_truthy | Mixed_non_maybe))),
-      LookupT (_, NonstrictReturning t_opt, [], _, _) ->
+    | (ObjProtoT reason |
+       FunProtoT reason |
+       DefT (reason, MixedT (Mixed_truthy | Mixed_non_maybe))),
+      LookupT (reason_op, NonstrictReturning t_opt, [], _, _) ->
```

Comment:
We will probably need to add a second parameter to `NonstrictReturning` or another variant to that data structure that lets us know we are coming from `TestPropT`. We use `NonstrictReturning` in a few other places that it would not make sense to add a lint warning.